### PR TITLE
VirtualNetworkGateway.json QOL fixes

### PR DESCRIPTION
**Pull request by henry416**
PR exclusively for general QOL fixes for VirtualNetworkGateway.json

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: arm-network/2017-06-01/swagger/virtualNetworkGateway.json [hunk 1]
```diff
@@ -1046,7 +976,17 @@
         "provisioningState": {
           "readOnly": true,
           "type": "string",
-          "description": "The provisioning state of the VirtualNetworkGateway resource. Possible values are: 'Updating', 'Deleting', and 'Failed'."
+          "description": "The provisioning state of the VirtualNetworkGateway resource.",
+          "enum": [
+            "Succeeded",
```
Comment (dsgouda) [inline arm-network/2017-06-01/swagger/virtualNetworkGateway.json]: Potential breaking change, please verify if that's okay

**Comment by henry416** [comment]
@dsgouda Normalized provisioning state across entire JSON

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
@azuresdkci Test this please

**Comment by salameer** [comment]
@dsgouda 
are we good t o merge here?

**Comment by dsgouda** [comment]
We are good, but I need to check why the travis build is failing, will investigate now.

**Comment by dsgouda** [comment]
CI is reporting a failure to generate sdk for ruby, get a feeling it may have to do with using old command line args for AutoRest, will post my findings soon.

**Comment by dsgouda** [comment]
@henry416 looks like you are updating `provisioningState` enum only here and not in the other json files which are a part of the corresponding composite. When generating the SDK, AutoRest expects to have exactly one unique `x-ms-enum` extension or the exact same definition repeated (like you have in virtualNetworkGateway.json), either update all definitions for `ProvisioningState` or undo this particular change. FYI The definition for `ProvisioningState` in networkWatcher.json has an additional enum value.

**Review by dsgouda — CHANGES_REQUESTED** [review]
enums need to be updated.

**Comment by salameer** [comment]
ping @henry416

**Comment by henry416** [comment]
@salameer I am aware of this, however I am busy with other issues at the moment, I will take time when it is finished to deal with this.

**Comment by salameer** [comment]
Sounds Good @henry416 

But please note that we'll have to close this pr by end of the Next week July 7th if this is not updated. and please feel free to open a new PR after that if your unable to make these changes by that time.

Thanks,

Samer

**Comment by salameer** [comment]
Closing due to no response.

**Pull request closed without merging**